### PR TITLE
Table component: new features for Eurovision sprint

### PR DESCRIPTION
**Pull request by javfg**
This PR includes the following:

- Global filter
- Configurable width
- Fixed height when using nowrap
- Fixed columns
- Fixed rows
- Server side pagination
- Server side sorting/filtering
- Documented the API

You can take a look at the Known Drugs or Related Diseases widgets in [here](https://5ece793e57f6270006b136db--platform-app.netlify.app/disease/EFO_0000384) to see the tables. Altough not all the functionality is enabled in those yet.

### CODE REVIEW

**Comment by javfg** [comment]
Just made a small change ([b2d4074](https://github.com/opentargets/platform-app/pull/155/commits/b2d4074caafdf791c4023d08c21c03483f64a03b))  on React keys for rows and header cells to make sure they don't duplicate.

**Comment by javfg** [comment]
Besides fixing the two problems Luca pointed out (commits e8a0571 and 4df5813), I fixed the data download widget inside the table; and changed the way the id field works in the column object.

To fix the downloader, I've included the component code inside table, because it is not used anywhere else, and it will be moved to ot-ui once we move the whole table.

On that process I detected a bug inside the downloader (it was not working when the id was a path); and added some features to it: you can now skip columns in the export (allowing us to export data from the related diseases table, for example), and it takes some defaults for headers.

Everything is detailed in the [table API doc](https://github.com/opentargets/platform-app/blob/jf-global-table_refactor-eurovision/src/public/common/Table/README.md).

File: src/public/common/Table/sortingAndFiltering.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const columnComparator = column?.comparator
+    ? column.comparator
+    : ascendingComparator;
+
+  return order === 'asc'
+    ? (a, b) => columnComparator(a, b, sortBy)
+    : (a, b) => -columnComparator(a, b, sortBy);
+}
+
+export function stableSort(array, comparator) {
```
Comment (mirandaio) [inline src/public/common/Table/sortingAndFiltering.js:22]: The sort in the standard library would work as well. It is ok to remove this sorting function

Comment (javfg) [reply src/public/common/Table/sortingAndFiltering.js]: Done!

**Pull request merged**